net/dnscache/dnscache.go

```diff
@@ -108,6 +112,19 @@ var debug = envknob.Bool("TS_DEBUG_DNS_CACHE")
 // If err is nil, ip will be non-nil. The v6 address may be nil even
 // with a nil error.
 func (r *Resolver) LookupIP(ctx context.Context, host string) (ip, v6 net.IP, allIPs []net.IPAddr, err error) {
+	if r.AllHostStaticResult != nil {
```

Comment:
check for host string as well here and return an error if it doesn't match, or panic

Reply:
I guess it says `AllHost`, fine as is

Reply:
that's not something we can do here. There is no specific host that's preconfigured on resolver.

Reply:
I meant something like `SingleHostStaticResult []netaddr.IP` and `SingleHostName []string`

Reply:
Done.